test/dash/dash_parser_content_protection_unit.js

```diff
@@ -299,6 +300,48 @@ describe('DashParser ContentProtection', () => {
     await testDashParser(source, expected);
   });
 
+  it('extracts embedded PSSHs with mspr:pro', async () => {
+    const source = buildManifestText([
+      // AdaptationSet lines
+      '<ContentProtection',
+      '  schemeIdUri="urn:uuid:9a04f079-9840-4286-ab92-e65be0885f95">',
+      '  <mspr:pro>ZmFrZSBQbGF5cmVhZHkgUFJP</mspr:pro>',
+      '</ContentProtection>',
+    ], [], []);
+
+    const proText = 'ZmFrZSBQbGF5cmVhZHkgUFJP';
+    const data = shaka.util.Uint8ArrayUtils.fromBase64(proText);
+    const systemId = new Uint8Array([
+      0x9a, 0x04, 0xf0, 0x79, 0x98, 0x40, 0x42, 0x86,
+      0xab, 0x92, 0xe6, 0x5b, 0xe0, 0x88, 0x5f, 0x95,
+    ]); // Playready
+    const pssh = shaka.util.Pssh.getPSSHFromMsprPro(data, systemId);
```

Comment:
Since the real data doesn't matter, you could have the `<mspr:pro>` element contain only one or two bytes and then inline the `pssh` data as a base64 string.  It would be better to avoid using this method since it is effectively the code under test.